core/src/main/java/org/ehcache/PersistentUserManagedEhcache.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright Terracotta, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.ehcache;
+
+import org.ehcache.config.ResourceType;
+import org.ehcache.events.CacheEventNotificationService;
+import org.ehcache.exceptions.CachePersistenceException;
+import org.ehcache.spi.cache.Store;
+import org.ehcache.spi.loaderwriter.CacheLoaderWriter;
+import org.ehcache.spi.service.LocalPersistenceService;
+import org.slf4j.LoggerFactory;
+
+/**
+ * PersistentUserManagedEhcache
+ */
+public class PersistentUserManagedEhcache<K, V> extends Ehcache<K, V> implements PersistentUserManagedCache<K, V> {
+
+  private final Store.PersistentStoreConfiguration storeConfig;
+  private final LocalPersistenceService localPersistenceService;
+  private final String id;
+
+  public PersistentUserManagedEhcache(RuntimeConfiguration<K, V> runtimeConfiguration, Store<K, V> store, Store.PersistentStoreConfiguration storeConfig, LocalPersistenceService localPersistenceService, CacheLoaderWriter<? super K, V> cacheLoaderWriter, CacheEventNotificationService<K, V> eventNotifier, String id) {
+    super(runtimeConfiguration, store, cacheLoaderWriter, eventNotifier, true, LoggerFactory.getLogger(PersistentUserManagedEhcache.class.getName() + "-" + id));
+    this.storeConfig = storeConfig;
+    this.localPersistenceService = localPersistenceService;
+    this.id = id;
+
+  }
+
+  @Override
+  public Maintainable toMaintenance() {
+    final StatusTransitioner.Transition st = internalToMaintenance();
+    try {
+      final Maintainable maintainable = new Maintainable() {
+        @Override
+        public void create() {
+          PersistentUserManagedEhcache.this.create();
+        }
+
+        @Override
+        public void destroy() {
+          PersistentUserManagedEhcache.this.destroy();
+        }
+
+        @Override
+        public void close() {
+          internalExitMaintenance().succeeded();
+        }
+      };
+      st.succeeded();
+      return maintainable;
+    } catch (RuntimeException e) {
+      st.failed(e); // this throws
```

Comment:
Can we not just change the contract of failed(t) to return the `StatusTranstionException` rather than throwing it?  That way this becomes:

```java
    return maintainable;
  } catch (RuntimeException e) {
    throw st.failed(e);
  }
}
```